src/finch/tensor.py

```diff
@@ -46,41 +46,72 @@ def size(self) -> int:
         return np.prod(self.shape)
 
     def __pos__(self):
-        return Tensor(jl_data=jl.Base.broadcast(jl.seval("+"), self._obj))
+        return Tensor(jl_data=jl.Base.broadcast(jl.seval("+"), self._get_obj()))
 
     def __add__(self, other):
-        return Tensor(jl_data=jl.Base.broadcast(jl.seval("+"), self._obj, other._obj))
+        return Tensor(jl_data=jl.Base.broadcast(jl.seval("+"), self._get_obj(), other._get_obj()))
 
     def __mul__(self, other):
-        return Tensor(jl_data=jl.Base.broadcast(jl.seval("*"), self._obj, other._obj))
+        return Tensor(jl_data=jl.Base.broadcast(jl.seval("*"), self._get_obj(), other._get_obj()))
 
     def __sub__(self, other):
-        return Tensor(jl_data=jl.Base.broadcast(jl.seval("-"), self._obj, other._obj))
+        return Tensor(jl_data=jl.Base.broadcast(jl.seval("-"), self._get_obj(), other._get_obj()))
 
     def __truediv__(self, other):
-        return Tensor(jl_data=jl.Base.broadcast(jl.seval("/"), self._obj, other._obj))
+        return Tensor(jl_data=jl.Base.broadcast(jl.seval("/"), self._get_obj(), other._get_obj()))
 
     def todense(self) -> np.ndarray:
-        shape = jl.size(self._obj)
+        obj = self._get_obj()
+        shape = jl.size(obj)
 
         # create fully dense array and access `val`
-        dense_lvls = jl.Element(jl.default(self._obj))
+        dense_lvls = jl.Element(jl.default(obj))
         for _ in shape:
             dense_lvls = jl.Dense(dense_lvls)
-        dense_tensor = jl.Tensor(dense_lvls, self._obj).lvl
+        dense_tensor = jl.Tensor(dense_lvls, obj).lvl
         for _ in shape:
             dense_tensor = dense_tensor.lvl
 
         result = np.array(dense_tensor.val).reshape(shape, order="F")
         return np.array(result, order="C")
 
+    def _get_obj(self) -> jc.AnyValue:
+        return self._get_materialized_obj() if self._is_swizzle() else self._obj
+
+    def _is_swizzle(self) -> bool:
+        return hasattr(self._obj, "body")
+
+    def _get_materialized_obj(self) -> "Tensor":
```

Comment:
I don't think this should be an option, I think we should view `swizzle(::Finch.Tensor)` as a materialized object and if the user wants to change the order they can copy into a different `Tensor` Object.

Reply:
That said, you can achieve this with `jl.copy(self._obj)`.

Reply:
Sure! I removed `_get_materialized_obj`.